api/src/main/java/io/onetable/client/HudiSourceConfig.java

```diff
@@ -0,0 +1,62 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+ 
+package io.onetable.client;
+
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.List;
+import java.util.Locale;
+
+import lombok.Value;
+
+import io.onetable.model.schema.PartitionTransformType;
+import io.onetable.spi.extractor.SourcePartitionSpecExtractor;
+
+public interface HudiSourceConfig {
```

Comment:
Should we have a more generic `SourceConfig` interface and then have a HudiSourceConfig as an implementation? I've also wanted to remove this class and auto-infer this information so mainly curious if you see value in an interface like `SourceConfig`

Reply:
Yes, I was going to revisit this one after some comments. Wasn't sure if HudiSourceConfig was somehow understood across formats and it had to be clear that it was the Hudi type. Again, I find the PerTableConfig with such a mix of format specifics odd. I need more info to understand why it can't all be consolidated into a OneTableConfig that extends the Hadoop Configuration object.

Reply:
We can aim to pull the pieces that are format specific into the source/target client. PerTableConfigs can hopefully just end up with information about the table itself.

I don't really want to bring Hadoop into this anymore than we have to so I'd rather not create config objects that extend Hadoop configuration.

Reply:
> I don't really want to bring Hadoop into this anymore than we have to so I'd rather not create config objects that extend
> Hadoop configuration.

This is an interesting statement. I believe that there is a Hadoop Configuration object passed to all clients created by the factory already. Are you planning to remove that?

Reply:
I am mainly trying to avoid extending Hadoop objects in case there are formats that do not require it. We can have hadoop configuration be a field but I would avoid extending. Prefer composition over inheritance in this case.

Reply:
@the-other-tim-brown  - I see. In that case, I should revisit the signature of the init method since it currently takes a perTableConfig and Configuration object. I usually make them no-arg methods but it seemed safe here to burn them in. I'll add some setters that will need to be called after instantiation and before init(). We'll need to add some simple reflection based injection. That can be a follow on, I think. Thanks for explaining the goal here.

Reply:
For the tasks that are still pending in this discussion, please create issues for them.